apps/frontend/src/pages/access-management/components/group-create-form/group-create-form.tsx

```diff
@@ -0,0 +1,98 @@
+import { Button, Input, Loader, Select } from "~/libs/components/components.js";
+import { DataStatus } from "~/libs/enums/enums.js";
+import {
+	useAppDispatch,
+	useAppForm,
+	useAppSelector,
+	useCallback,
+	useEffect,
+	useSelectedItems,
+} from "~/libs/hooks/hooks.js";
+import {
+	type GroupCreateRequestDto,
+	groupCreateValidationSchema,
+} from "~/modules/groups/groups.js";
+import { actions as permissionActions } from "~/modules/permissions/permissions.js";
+
+import { GroupUsersTable } from "../group-users-table/group-users-table.js";
+import { DEFAULT_GROUP_CREATE_PAYLOAD } from "./libs/constants/constants.js";
+import { getPermissionOptions } from "./libs/helpers/helpers.js";
+import styles from "./styles.module.css";
+
+type Properties = {
+	onSubmit: (payload: GroupCreateRequestDto) => void;
+};
+
+const GroupCreateForm = ({ onSubmit }: Properties): JSX.Element => {
+	const dispatch = useAppDispatch();
+	const { control, errors, handleSubmit, setValue } =
+		useAppForm<GroupCreateRequestDto>({
+			defaultValues: DEFAULT_GROUP_CREATE_PAYLOAD,
+			validationSchema: groupCreateValidationSchema,
+		});
+
+	const { dataStatus: permissionsDataStatus, permissions } = useAppSelector(
+		({ permissions }) => permissions,
+	);
+
+	useEffect(() => {
+		void dispatch(permissionActions.loadAll());
+	}, [dispatch]);
+
+	const handleFormSubmit = useCallback(
+		(event_: React.BaseSyntheticEvent): void => {
+			void handleSubmit((formData: GroupCreateRequestDto) => {
+				onSubmit({
+					name: formData.name,
+					permissionIds: formData.permissionIds,
+					userIds: formData.userIds,
+				});
+			})(event_);
+		},
+		[handleSubmit, onSubmit],
+	);
+
+	const permissionOptions = getPermissionOptions(permissions);
+
+	const isLoading =
+		permissionsDataStatus === DataStatus.IDLE ||
+		permissionsDataStatus === DataStatus.PENDING;
+
+	const { items: selectedUserIds, onToggle: handleToggle } =
```

Comment:
let's name it consistently
```suggestion
	const { items: selectedUserIds, onToggle: handleUserIdsToggle } =
```